packages/core/core/src/AssetGraphBuilder.js

```diff
@@ -64,29 +74,37 @@ export default class AssetGraphBuilder extends EventEmitter {
     this.graph.initializeGraph({entries, targets, transformerRequest, rootDir});
 
     this.controller = new AbortController();
-    if (options.watch) {
-      this.watcher = new Watcher();
-      this.watcher.on('change', async filePath => {
-        if (this.graph.hasNode(filePath)) {
-          this.controller.abort();
-          this.graph.invalidateFile(filePath);
-
-          this.emit('invalidate', filePath);
-        }
-      });
-    }
   }
 
-  async initFarm() {
+  async init() {
     // This expects the worker farm to already be initialized by Parcel prior to calling
     // AssetGraphBuilder, which avoids needing to pass the options through here.
     this.farm = await WorkerFarm.getShared();
     this.runTransform = this.farm.mkhandle('runTransform');
+
+    if (this.options.watch) {
+      let ignore = ['.git', '.hg', '.parcel-cache'].map(dir =>
+        path.join(this.rootDir, dir)
+      );
+      this.watcher = await watcher.subscribe(
+        this.rootDir,
```

Comment:
How do you see multiple watch roots working? Would core have to manage multiple subscriptions or should the subscription be able to handle multiple paths?